In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('Indian Airlines.csv')

In [3]:
df.duplicated().value_counts()

False    300153
Name: count, dtype: int64

In [4]:
%load_ext sql
%sql duckdb://

Connecting to 'duckdb://'

### Understanding the dataset

In [5]:
## Total rows & columns
display(print("Total rows: {} \nTotal columns: {}".format(df.shape[0], df.shape[1])))

## Count for missing values in the dataset
display(print('Total NaN values in dataset: {}'.format(df.isnull().values.sum())))

## Drop duplicate rows from the dataset
display(print("The dataset has {} duplicate rows".format(len(df) - len(df.drop_duplicates()))))
# df.drop_duplicates(inplace=True) - execute this command if there are duplicates

## Basic info of the entire dataset
display(df.info())

Total rows: 300153 
Total columns: 12


None

Total NaN values in dataset: 0


None

The dataset has 0 duplicate rows


None

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300153 entries, 0 to 300152
Data columns (total 12 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Unnamed: 0        300153 non-null  int64  
 1   airline           300153 non-null  object 
 2   flight            300153 non-null  object 
 3   source_city       300153 non-null  object 
 4   departure_time    300153 non-null  object 
 5   stops             300153 non-null  object 
 6   arrival_time      300153 non-null  object 
 7   destination_city  300153 non-null  object 
 8   class             300153 non-null  object 
 9   duration          300153 non-null  float64
 10  days_left         300153 non-null  int64  
 11  price             300153 non-null  int64  
dtypes: float64(1), int64(3), object(8)
memory usage: 27.5+ MB


None

##### --------------------------------------------------------------------------------------------------------------------------
Conculsion: 
1. The dataset has no NaN / empty / null / None values
2. Dataset has total 300,153 rows & 12 columns
3. Dataset has 0 duplicates


### Drop Unnecessary columns (optional)

In [6]:
## Drop multiple columns ('Unnamed: 0', 'flight') that are not required
drop_cols = [0,2]

df.drop(df.columns[drop_cols], axis = 1, inplace = True)

df.head()

airline source_city departure_time stops   arrival_time destination_city  \
0  SpiceJet       Delhi        Evening  zero          Night           Mumbai   
1  SpiceJet       Delhi  Early_Morning  zero        Morning           Mumbai   
2   AirAsia       Delhi  Early_Morning  zero  Early_Morning           Mumbai   
3   Vistara       Delhi        Morning  zero      Afternoon           Mumbai   
4   Vistara       Delhi        Morning  zero        Morning           Mumbai   

     class  duration  days_left  price  
0  Economy      2.17          1   5953  
1  Economy      2.33          1   5953  
2  Economy      2.17          1   5956  
3  Economy      2.25          1   5955  
4  Economy      2.33          1   5955

In [7]:
%%sql
SELECT 
* 
From df

Running query in 'duckdb://'

+----------+-------------+----------------+-------+---------------+------------------+---------+----------+-----------+-------+
| airline  | source_city | departure_time | stops |  arrival_time | destination_city |  class  | duration | days_left | price |
+----------+-------------+----------------+-------+---------------+------------------+---------+----------+-----------+-------+
| SpiceJet |    Delhi    |    Evening     |  zero |     Night     |      Mumbai      | Economy |   2.17   |     1     |  5953 |
| SpiceJet |    Delhi    | Early_Morning  |  zero |    Morning    |      Mumbai      | Economy |   2.33   |     1     |  5953 |
| AirAsia  |    Delhi    | Early_Morning  |  zero | Early_Morning |      Mumbai      | Economy |   2.17   |     1     |  5956 |
| Vistara  |    Delhi    |    Morning     |  zero |   Afternoon   |      Mumbai      | Economy |   2.25   |     1     |  5955 |
| Vistara  |    Delhi    |    Morning     |  zero |    Morning    |      Mumbai      | Economy |   2.33   |     1     |  5955 |
| Vistara  |    Delhi    |    Morning     |  zero |   Afternoon   |      Mumbai      | Economy |   2.33   |     1     |  5955 |
| Vistara  |    Delhi    |    Morning     |  zero |    Morning    |      Mumbai      | Economy |   2.08   |     1     |  6060 |
| Vistara  |    Delhi    |   Afternoon    |  zero |    Evening    |      Mumbai      | Economy |   2.17   |     1     |  6060 |
| GO_FIRST |    Delhi    | Early_Morning  |  zero |    Morning    |      Mumbai      | Economy |   2.17   |     1     |  5954 |
| GO_FIRST |    Delhi    |   Afternoon    |  zero |    Evening    |      Mumbai      | Economy |   2.25   |     1     |  5954 |
+----------+-------------+----------------+-------+---------------+------------------+---------+----------+-----------+-------+
Truncated to displaylimit of 10.